server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java

```diff
@@ -349,13 +342,6 @@ private void submitStateUpdateTask(final String source, final ClusterStateTaskCo
         }
     }
 
-    /** asserts that the current thread is the cluster state update thread */
-    public static boolean assertClusterStateUpdateThread() {
```

Comment:
I am looking into the history of why this method is no longer used. We definitely use to assert this from, for example, `ZenDiscovery` at least as recently as the 5.6 days.

Reply:
Okay, this makes sense. It happened when we separated the cluster applier service from the master service.